markups/commonmark.py

```diff
@@ -0,0 +1,56 @@
+# vim: ts=8:sts=8:sw=8:noexpandtab
+
+# This file is part of python-markups module
+# License: BSD
+# Copyright: (C) Shengjing Zhu, 2016
+
+import markups.common as common
+from markups.abstract import AbstractMarkup, ConvertedMarkup
+
+
+class CommonMarkMarkup(AbstractMarkup):
+	"""Markup class for CommonMark language.
+	Inherits :class:`~markups.abstract.AbstractMarkup`.
+	"""
+	name = 'CommonMark'
+	attributes = {
+		common.LANGUAGE_HOME_PAGE: 'http://commonmark.org/',
+		common.MODULE_HOME_PAGE: 'https://github.com/rtfd/CommonMark-py/',
+		common.SYNTAX_DOCUMENTATION: 'http://spec.commonmark.org/'
+	}
+
+	# Note: the same as MarkdownMarkup
+	# file_extensions = ('.md', '.mkd', '.mkdn', '.mdwn', '.mdown', '.markdown')
+	# default_extension = '.md'
+
+	@staticmethod
+	def available():
+		try:
+			import CommonMark
+		except ImportError:
+			return False
+		return True
+
+	def __init__(self, filename=None, options={}):
+		self.enable_highlight = options.get('highlight', False)
+		super().__init__(filename)
+		import CommonMark
+		from markups.cmark_render import CommonMarkRender
+		self.parse = CommonMark.Parser().parse
+		if self.enable_highlight:
```

Comment:
I do not want to add any code to ReText specific to this markup. Is there any reason to not use Pygments when it is available?